### ISSUE TITLE
Agenda for pymbar planning call on Jan 6th, 12 EST/10 MST 

### ISSUE THREAD

**Issue by mrshirts** [choderalab/pymbar#254]
Please edit and add things:

* What are the finishing bits to have a 3.0.0 release that?
   - Runs CI
   - is compatible with python 2.0 and 3.0.
   - Solves at least all the issues encountered up through the release of 2.1.
   - includes bugfixes for robustness that were introduced since 2.1

* What is the plan moving forward?

 phase 1: Robust 'adaptive' solver supporting both u_nk and u_kln input formats, works with python 2.x and 3.x (MRS: should be in 3.0.0 release)

 phase 2: Compiled extension to speed up 'adaptive' solvers

 phase 3: Fast alternative solvers for big datasets, maybe accessible under another API point (such  as pymbar.densesolvers). We can make these default if we have automatic fallback if they fail to converge.

 phase 4: Fast alternative solvers for large datasets that use sparse structure, such as using Zhiqiang's new sparse or stochastic solvers.  (MRS note: the standard adaptive solvers are faster than stochastic solver for essentially any matrix that can be stored in 16 GB. ZT's examples used a dataset where the reduced potential energies were calculated on the fly -- otherwise it wouldn't fit into memory.  I would love to look at the sparse solver more, I'm still trying to figure out how to implement it for more than nearest neighbor).

* What is worth salvaging from the 3.0 code?
  - Speedups for large data sets?  How can we get these with minimal code perturbation so that 

* How can we best incorporate contributed datasets with CI?

* Should we separate out the timeseries code, since it's a modular component useful for other things?

 * useful features that should be supported eventually (to be implemented contingent on manpower).
  - calculation of covariance for iterated differences.

**Comment by jchodera** [comment]
Speedups are irrelevant if the code does not work or is not robust.

Our primary focus *needs* to be: "How do we get to a robust python 2.x and 3.x compatible code that works and can be continuously tested?" 

We don't have that yet, and we need that first. We can discuss all of the other things after we formulate a plan to get to this point.

**Comment by mrshirts** [comment]
> We don't have that yet, and we need that first. We can discuss all of the other things after we formulate a plan to get to this point. 

You said "If your group has the time, manpower, expertise, and bandwidth to get it ready for release, you can present a plan for doing so at the meeting". So it seemed like that was supposed t be part of the meeting.

Clearly getting python 2.x and 3.x compatible code that can be tested is first.

**Comment by jchodera** [comment]
> Clearly getting python 2.x and 3.x compatible code that can be tested is first.

Great, sounds like we're on the same page here.

**Comment by Lnaden** [comment]
Superseded by #255
